src/global_distrib/mod_global_distrib_mapping.erl

```diff
@@ -298,3 +302,42 @@ do_lookup_jid({_, _, _} = Jid) ->
                 BareJid -> ets_cache:lookup(?JID_TAB, BinJid, LookupInDB(jid:to_binary(BareJid)))
             end
     end.
+
+-spec get_advertised_endpoints(Opts :: [{Key :: term(), Value :: term()}]) -> [endpoint()].
+get_advertised_endpoints(Opts) ->
+    Conns = proplists:get_value(connections, Opts, []),
+    AdvEndps = proplists:get_value(advertised_endpoints, Conns),
+    case AdvEndps of
+        undefined ->
+            false;
+        Endpoints ->
+            {WithIp, WithDomain} = endpoints_with_ipaddr_and_domains(Endpoints),
+            mod_global_distrib_utils:resolve_endpoints(WithIp) ++ parse_domains(WithDomain)
+    end.
+
+-spec endpoints_with_ipaddr_and_domains(AllEndpoints :: [endpoint()])
+                                       -> {IpEndpoints :: [endpoint()], DomainEndpoints :: [endpoint()]}.
+endpoints_with_ipaddr_and_domains(Endpoints) ->
+    WithDomain = lists:filter(fun({Domain, _Port}) ->
+                                        is_domain(Domain) end, Endpoints),
+    WithIp = Endpoints -- WithDomain,
+    {WithIp, WithDomain}.
+
+-spec is_domain(any()) -> boolean().
+is_domain(Domain) when is_tuple(Domain) -> false;
```

Comment:
@fenek or @arcusfelis, do you think this function name is ok? I think it might be ambiguous but can't think of a nice and short name.

Reply:
Especially that it might soon go out of this module to utils :p